File: tests/mozmill/tests/testLinks/testTextSelection/testOpenInCurrentTab.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

var {assert, expect} = require("../../../../../../../lib/assertions");
var prefs = require("../../../../../../lib/prefs");
var tabs = require("../../../../../../lib/tabs");

var rpUtils = require("../../../lib/rp-utils");
var rpConst = require("../../../lib/constants");

const TEST_URL = "http://www.maindomain.test/link_1.html";


var setupModule = function(aModule) {
  aModule.controller = mozmill.getBrowserController();
  aModule.tabBrowser = new tabs.tabBrowser(aModule.controller);
  aModule.tabBrowser.closeAllTabs();

  prefs.preferences.setPref(rpConst.PREF_DEFAULT_ALLOW, false);
}

var teardownModule = function(aModule) {
  prefs.preferences.clearUserPref(rpConst.PREF_DEFAULT_ALLOW);
  aModule.tabBrowser.closeAllTabs();
}


var testOpenInCurrentTab = function() {
  var tabIndex = tabBrowser.selectedIndex;

  controller.open(TEST_URL);
  controller.waitForPageLoad();

  let textURL = findElement.ID(controller.window.document, "text_url_1");
  let textURLValue = textURL.getNode().textContent;

  rpUtils.selectText(tabBrowser, textURLValue);

  // perform right-click and entry selection
  var contextMenu = controller.getMenu("#contentAreaContextMenu");
  contextMenu.select("#context-openlinkincurrent", textURL);

  rpUtils.waitForTabLoad(controller, tabBrowser.getTab(tabIndex));

  var panel = tabBrowser.getTabPanelElement(tabIndex,
      '/{"value":"' + rpConst.REDIRECT_NOTIFICATION_VALUE + '"}');
  assert.ok(false === panel.exists(),
      "Following the URL didn't cause a redirect");

  assert.equal(controller.tabs.activeTab.location.href, textURLValue,
      "The location is correct.");
}

File: tests/marionette/tests/links/text_selection/test_open_in_current_tab.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from rp_ui_harness import RequestPolicyTestCase


TEST_URL = "http://www.maindomain.test/link_1.html";
PREF_DEFAULT_ALLOW = "extensions.requestpolicy.defaultPolicy.allow"


class TestOpenInCurrentTab(RequestPolicyTestCase):

    def setUp(self):
        RequestPolicyTestCase.setUp(self)
        self.prefs.set_pref(PREF_DEFAULT_ALLOW, False);


    def test_open_in_current_tab(self):
        with self.marionette.using_context("content"):
            # load the test url
            self.marionette.navigate(TEST_URL)

            # find the URL string and its wrapping element
            url_wrapper = self.marionette.find_element("id", "text_url_1")
            url = url_wrapper.get_attribute("textContent")

            # select the URL
            self.web_utils.select_element_text(url_wrapper)

        with self.marionette.using_context("chrome"):
            # perform right-click and entry selection
            self.ctx_menu.select_entry("context-openlinkincurrent", url_wrapper)

            tab = self.browser.tabbar.selected_tab
            self.tabs.wait_until_loaded(tab)

            self.assertFalse(self.redir.panel_exists(),
                             "Following the URL didn't cause a redirect.")
            self.assertEqual(tab.location, url,
                             "The location is correct.")
